shared/teams/team/rows/index.tsx

```diff
@@ -167,3 +172,97 @@ export const useSubteamsSections = (
   }
   return sections
 }
+
+const useGeneralConversationIDKey = (teamID?: Types.TeamID) => {
+  const [conversationIDKey, setConversationIDKey] = React.useState<Chat2Types.ConversationIDKey | null>(null)
+  const generalConvID = Container.useSelector(
+    (state: Container.TypedState) => teamID && state.chat2.teamIDToGeneralConvID.get(teamID)
+  )
+  const dispatch = Container.useDispatch()
+  React.useEffect(() => {
+    if (!conversationIDKey && teamID) {
+      if (!generalConvID) {
+        dispatch(Chat2Gen.createFindGeneralConvIDFromTeamID({teamID}))
+      } else {
+        setConversationIDKey(generalConvID)
+      }
+    }
+  }, [conversationIDKey, dispatch, generalConvID, teamID])
+  return conversationIDKey
+}
+
+export const useEmojiSections = (teamID: Types.TeamID): Array<Section> => {
+  const convID = useGeneralConversationIDKey(teamID)
+  const getUserEmoji = Container.useRPC(RPCChatTypes.localUserEmojisRpcPromise)
+  const [customEmoji, setCustomEmoji] = React.useState<RPCChatTypes.Emoji[]>([])
+  const [waiting, setWaiting] = React.useState(true)
+
+  const [filter, setFilter] = React.useState('')
+
+  React.useEffect(() => {
```

Comment:
I'd rather not fire off RPCs if we're not on the emoji tab - see `useChannelsSections` for an example of delaying it if we're not on that tab.

And if you think there's a better general way of doing it than that, feel free to say so, that's just the way I came up with at the time.